File: Codeforces-160A.cpp
#include <iostream>
#include <string>
#include <vector>
#include <map>
#include <algorithm>
 
using namespace std;
 
int main()
{
    int n,sum=0,suml=0,j=0;
    cin >> n;
    vector<int>v(n);
    for (int i = 0; i < n; i++) {
        cin >> v[i];
        sum += v[i];
    }
    sort(v.begin(), v.end());
    if (v.size() == 2) {
        if (v[0] == v[1]) {
            cout << "2";
        }
        else {
            cout << "1";
        }
    }
    else {
        for (int i = n - 1; i >= 0; i--) {
 
            if (suml <= (sum / 2)) {
                suml += v[i];
                j++;
            }
        }
        cout << j << endl;
    }
 
 
    return 0;
}

File: Codeforces-160A.java


import java.util.ArrayList;
import java.util.Arrays;
import java.util.Scanner;


public class Main{
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);

        int n = in.nextInt();
        int sum=0;
        int[] arr = new int[n];

        for(int i=0; i<n; i++){
            arr[i]=in.nextInt();
            sum+=arr[i];
        }
        if(n==2){
            if(arr[0]==arr[1]) {
                System.out.println("2");
            }else{
                System.out.println("1");
            }
        }else {
            Arrays.sort(arr);
            int coinSum=0;
            int coins=0;
            for(int i=n-1; i>=0; i--){
                if(coinSum<=(sum/2)){
                    coinSum+=arr[i];
                    coins++;
                }
            }
            System.out.println(coins);
        }
        in.close();
    }
}